core/src/main/scala/org/bitcoins/core/protocol/dlc/ContractInfo.scala

```diff
@@ -137,17 +138,41 @@ case class ContractInfo(
 
   /** Map OracleOutcomes (which correspond to CETs) to their adpator point and payouts */
   lazy val outcomeMap: Map[OracleOutcome, (ECPublicKey, Satoshis, Satoshis)] = {
-    val builder =
-      HashMap.newBuilder[OracleOutcome, (ECPublicKey, Satoshis, Satoshis)]
+    Await.result(
```

Comment:
Opinions of changing the return type to `Future[Map[...]]`? 

Also we probably want to give the abillity to passing in a custom execution context rather than running on the scala runtime execution context. This would mean we have to maek a choice between

1. Making this a method, and build in caching so we can provide the execution context
2. Make the execution context a class level parameter.

Reply:
I'm personally very hesitant towards starting to hyper-optimize and `Future`-ify this code without first being clear about how this library is going to work in the future (e.g. will there be two versions of dlc code where one is really fast using akka and the other is synchronous and simpler?)